src/catalogue/configuration/db/type-orm-entity/produitTypeORMEntity.ts

```diff
@@ -0,0 +1,37 @@
+import {Column, Entity, PrimaryColumn} from "typeorm";
+import Produit from "../../../domain/entities/produit";
+
+@Entity({name: 'produits'})
+export default class ProduitTypeORMEntity {
+	@PrimaryColumn()
+	readonly id: string
+
+	@Column()
+	readonly nom: string
+
+	@Column({type: 'float'})
+	prix: number
+
+	@Column({type: 'float'})
+	poids: number
+
+	constructor(id: string, nom: string, prix: number, poids: number) {
+		this.id = id
```

Comment:
On peut pas définir l'id ici plutôt ? Genre this.id = new UUID() ?

Reply:
Je ne pense pas parce que :
 - l'identifiant du produit est une notion métier, donc c'est bien que sa création apparaisse dans le use case
 - pour suivre le principe d'injection de dépendance, la tâche de créer un id appartient à la classe UUIDGenerator. SI demain on veut utiliser d'autres id, il suffit de modifier cette classe + ça simplifie les tests parce qu'on peut mocker/stubber cette classe

Reply:
Très clair suis d'accord